src/Storages/MessageQueueSink.cpp

```diff
@@ -60,21 +60,21 @@ void MessageQueueSink::consume(Chunk & chunk)
                 row_format->writeRow(columns, row);
             }
             row_format->finalize();
-            row_format->resetFormatter();
             producer->produce(buffer->str(), i, columns, row - 1);
             /// Reallocate buffer if it's capacity is large then DBMS_DEFAULT_BUFFER_SIZE,
             /// because most likely in this case we serialized abnormally large row
             /// and won't need this large allocated buffer anymore.
             buffer->restart(DBMS_DEFAULT_BUFFER_SIZE);
+            row_format->resetFormatter();
         }
     }
     else
     {
         format->write(getHeader().cloneWithColumns(chunk.detachColumns()));
         format->finalize();
         producer->produce(buffer->str(), chunk.getNumRows(), columns, chunk.getNumRows() - 1);
-        format->resetFormatter();
         buffer->restart();
+        format->resetFormatter();
```

Comment:
It would be good to add a comment on why this order of calls is the correct one. Or is it obvious (I'm just unfamiliar with this code)?